traffic_ops/testing/api/v4/deliveryservices_test.go

```diff
@@ -171,13 +173,278 @@ func cleanUp(t *testing.T, ds tc.DeliveryServiceV4, oldCDNID int, newCDNID int)
+	_, _, err = TOSession.GenerateSSLKeysForDS(*ds2.XMLID, *ds2.CDNName, tc.SSLKeyRequestFields{
+		BusinessUnit: util.StrPtr("BU"),
+		City:         util.StrPtr("CI"),
+		Organization: util.StrPtr("OR"),
+		HostName:     util.StrPtr("*.test2.com"),
+		Country:      util.StrPtr("CO"),
+		State:        util.StrPtr("ST"),
+	})
+	if err != nil {
+		t.Fatalf("unable to generate sslkeys for DS %v: %v", *customDS2.XMLID, err)
+	}
+
+	tries := 0
+	var cdnKeys []tc.CDNSSLKeys
+	for tries < 5 {
+		time.Sleep(time.Second)
+		cdnKeys, _, err = TOSession.GetCDNSSLKeys(cdn.Name, nil)
+		if err == nil && len(cdnKeys) != 0 {
+			break
+		}
+		tries++
+	}
+	if err != nil {
+		t.Fatalf("unable to get CDN %v SSL keys: %v", cdn.Name, err)
+	}
+	if len(cdnKeys) != 2 {
+		t.Errorf("expected two ssl keys for CDN %v, got %d instead", cdn.Name, len(cdnKeys))
+	}
+
+	_, err = TOSession.DeleteDeliveryService(*ds2.ID)
+	if err != nil {
+		t.Errorf("could not delete delivery service %v", *ds.XMLID)
+	}
+	_, _, err = TOSession.SnapshotCRConfigByID(cdn.ID)
+	if err != nil {
+		t.Fatalf("couldn't snap CDN: %v", err)
+	}
+	tries = 0
+	var newCdnKeys []tc.CDNSSLKeys
+	for tries < 5 {
+		time.Sleep(time.Second)
+		newCdnKeys, _, err = TOSession.GetCDNSSLKeys(cdn.Name, nil)
+		tries++
+	}
+	if err != nil {
+		t.Fatalf("unable to get CDN %v SSL keys: %v", cdn.Name, err)
+	}
+	if len(newCdnKeys) != 1 {
+		t.Errorf("expected 1 ssl keys for CDN %v, got %d instead", cdn.Name, len(newCdnKeys))
+	}
+}
+
+func DeliveryServiceSSLKeys(t *testing.T) {
+	cdn := createBlankCDN("sslkeytransfer2", t)
+
+	types, _, err := TOSession.GetTypeByName("HTTP", nil)
+	if err != nil {
+		t.Fatal("unable to get types: " + err.Error())
+	}
+	if len(types) < 1 {
+		t.Fatal("expected at least one type")
+	}
+
+	customDS := tc.DeliveryServiceV4{}
+	customDS.Active = util.BoolPtr(true)
+	customDS.CDNID = util.IntPtr(cdn.ID)
+	customDS.DSCP = util.IntPtr(0)
+	customDS.DisplayName = util.StrPtr("displayName2")
+	customDS.RoutingName = util.StrPtr("routingName2")
+	customDS.GeoLimit = util.IntPtr(0)
+	customDS.GeoProvider = util.IntPtr(0)
+	customDS.IPV6RoutingEnabled = util.BoolPtr(false)
+	customDS.InitialDispersion = util.IntPtr(1)
+	customDS.LogsEnabled = util.BoolPtr(true)
+	customDS.MissLat = util.FloatPtr(0)
+	customDS.MissLong = util.FloatPtr(0)
+	customDS.MultiSiteOrigin = util.BoolPtr(false)
+	customDS.OrgServerFQDN = util.StrPtr("https://test2.com")
+	customDS.Protocol = util.IntPtr(2)
+	customDS.QStringIgnore = util.IntPtr(0)
+	customDS.RangeRequestHandling = util.IntPtr(0)
+	customDS.RegionalGeoBlocking = util.BoolPtr(false)
+	customDS.TenantID = util.IntPtr(1)
+	customDS.TypeID = util.IntPtr(types[0].ID)
+	customDS.XMLID = util.StrPtr("dsID2")
+	customDS.MaxRequestHeaderBytes = nil
+
+	ds, _, err := TOSession.CreateDeliveryService(customDS)
+	if err != nil {
+		t.Fatal(err)
+	}
+	ds.CDNName = &cdn.Name
+	_, _, err = TOSession.GenerateSSLKeysForDS(*ds.XMLID, *ds.CDNName, tc.SSLKeyRequestFields{
```

Comment:
nothing to do here, just an observation: I have no idea why this method would need to take a CDN name. The CDN name can be inferred by the provided DS. I'll open a ticket to follow up on this.